compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -13019,11 +13131,15 @@ TR::Node *ificmpgeSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier
 
    makeConstantTheRightChildAndSetOpcode(node, firstChild, secondChild, s);
 
+   IfcmpgeReducer ifcmpReducer(node, s);
+
    if (node->getOpCodeValue() == TR::ificmpge)
       {
       if (firstChild->getOpCode().isLoadConst() && conditionalBranchFold((originalFirst->getInt() >= originalSecond->getInt()), node, firstChild, secondChild, block, s))
          return node;
       intCompareNarrower(node, s, TR::ifsucmpge, TR::ifscmpge, TR::ifbcmpge);
+      if (ifcmpReducer.reducible<int32_t>())
```

Comment:
The preceding call to `intCompareNarrower` can change the comparison opcode from `ificmpge` to one of `ifsucmpge`, `ifscmpge`, or `ifbcmpge`, in which case we'd want `reducible<uint16_t>`, `reducible<int16_t>`, or `reducible<int8_t>`, respectively.

It's called again in the `ifiucmpge` branch below, and `iflcmpgeSimplifier` has a similar call to `longCompareNarrower`.

I'd recommend making the existing `reducible` private and renaming it, and wrapping it with something like the switch you have below in `ifCmpWithEqualitySimplifier`, so that callers can simply ask `ifcmpReducer.reducible()`

Reply:
Fixed.